src/lib/y2partitioner/actions/controllers/lvm_vg.rb

```diff
@@ -0,0 +1,298 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  module Actions
+    module Controllers
+      # This class stores information about an LVM volume group being created and
+      # takes care of updating the devicegraph when needed
+      class LvmVg
+        include Yast::I18n
+
+        # @return [String] given volume group name
+        attr_accessor :vg_name
+
+        # @return [Y2Storage::DiskSize] given extent size
+        attr_reader :extent_size
+
+        # @return [Y2Storage::LvmVg] new created volume group
+        attr_reader :vg
+
+        DEFAULT_VG_NAME = "".freeze
+        DEFAULT_EXTENT_SIZE = Y2Storage::DiskSize.MiB(4).freeze
+
+        ALLOWED_NAME_CHARS =
+          "0123456789" \
+          "abcdefghijklmnopqrstuvwxyz" \
+          "ABCDEFGHIJKLMNOPQRSTUVWXYZ" \
+          "._-+".freeze
+
+        # Constructor
+        #
+        # @note This will create a new LVM volume group in the devicegraph right away.
+        # @see #initialize_values
+        def initialize
+          textdomain "storage"
+
+          initialize_values
+        end
+
+        # Stores the given extent size
+        # @note The value is internally stored as DiskSize (see {to_size}) when it can
+        #   be parsed; nil otherwise.
+        #
+        # @param value [String]
+        def extent_size=(value)
+          @extent_size = to_size(value)
+        end
+
+        # Effective size of the resulting LVM volume group
+        #
+        # @return [Y2Storage::DiskSize]
+        def vg_size
+          vg.size
+        end
+
+        # Applies given values (i.e., volume group name and extent size) to the
+        # volume group
+        def apply_values
+          vg.vg_name = vg_name
+          vg.extent_size = extent_size
```

Comment:
Assigning nil here to `extend_size` is correctly handled? Doesn't the library go nuts? Even more important, is that situation unit-tested?

Reply:
This method is only called when all validations are passed, so in practice it is not possible to assign a nil value.